src/test/java/com/fittracker/fittracker/controller/WeightControllerValidationTest.java

```diff
@@ -157,11 +158,145 @@ public void givenPostRequestWithNullDate_shouldReturnDateErrorMessage() throws E
+
+            verifyNoInteractions(weightService);
+        }
+
+        @ParameterizedTest
+        @CsvSource({
+                "-1, Value must be positive",
+                "700, Value must be less or equal to 635"
+        })
+        void givenPutRequest_shouldReturnValueErrorMessage(String value, String message) throws Exception {
+            mockMvc
+                    .perform(put(URI.create(ENDPOINT))
+                            .contentType(APPLICATION_JSON)
+                            .content(format("{\"date\":\"2023-08-03\",\"value\": %s}", value)))
+                    .andExpect(status().isBadRequest())
+                    .andExpect(content().string(format("{\"field\":\"value\",\"message\":\"%s\"}", message)));
+
+            verifyNoInteractions(weightService);
+        }
+
+        @Test
+        void givenPutRequestWithNullValue_shouldReturnValueErrorMessage() throws Exception {
+            mockMvc
+                    .perform(put(URI.create(ENDPOINT))
+                            .contentType(APPLICATION_JSON)
+                            .content("{\"date\":\"2023-08-03\"}"))
+                    .andExpect(status().isBadRequest())
+                    .andExpect(content().string("{\"field\":\"value\",\"message\":\"Value must not be null\"}"));
+
+            verifyNoInteractions(weightService);
+        }
+
+        @Test
+        public void givenPutRequestWithValidDate_shouldReturnOk() throws Exception {
+            mockMvc
+                    .perform(put(URI.create(ENDPOINT))
+                            .contentType(APPLICATION_JSON)
+                            .content("{\"date\":\"2023-10-10\",\"value\": 13.2}"))
+                    .andExpect(status().isOk())
+                    .andExpect(content().string(""));
+
+            verify(weightService).update(any());
+        }
+
+        @Test
+        public void givenPutRequestWithDateNow_shouldReturnOk() throws Exception {
+            mockMvc
+                    .perform(put(URI.create(ENDPOINT))
+                            .contentType(APPLICATION_JSON)
+                            .content(format("{\"date\":\"%s\",\"value\": 13.2}", LocalDate.now())))
+                    .andExpect(status().isOk())
+                    .andExpect(content().string(""));
+
+            verify(weightService).update(any());
+        }
+
+        @Test
+        public void givenPutRequestWithNullDate_shouldReturnDateErrorMessage() throws Exception {
+            mockMvc
+                    .perform(put(URI.create(ENDPOINT))
+                            .contentType(APPLICATION_JSON)
+                            .content("{\"value\": 13.2}"))
+                    .andExpect(status().isBadRequest())
+                    .andExpect(content().string("{\"field\":\"date\",\"message\":\"Date must not be null\"}"));
+
+            verifyNoInteractions(weightService);
+        }
+
+    }
+
+    @Nested
+    class Delete {
```

Comment:
these tests will need some updating once we change it to use query param (will be similar to GET)